src/software/utils/main_imageProcessing.cpp

```diff
@@ -120,128 +339,210 @@ int aliceVision_main(int argc, char * argv[])
-            oiio::ImageBuf outBuf(oiio::ImageSpec(image.Width(), image.Height(), nchannels, oiio::TypeDesc::FLOAT), filtered.data());
-            oiio::ImageBufAlgo::contrast_remap(outBuf, inBuf, 0.0f, 1.0f, 0.0f, 1.0f, contrast);
-            image.swap(filtered);
-    #endif
+            for (auto & i : sfmData.getIntrinsics())
+            {
+                i.second->rescale(pParams.downscale);
+            }
         }
-        if (medianFilter >= 3)
+
+        // Save sfmData with modified path to images
+        if (!sfmDataIO::Save(sfmData, sfmOutputDataFilepath, sfmDataIO::ESfMData(sfmDataIO::ALL)))
         {
-            image::Image<image::RGBfColor> filtered(image.Width(), image.Height());
-            const oiio::ImageBuf inBuf(oiio::ImageSpec(image.Width(), image.Height(), nchannels, oiio::TypeDesc::FLOAT), image.data());
-            oiio::ImageBuf outBuf(oiio::ImageSpec(image.Width(), image.Height(), nchannels, oiio::TypeDesc::FLOAT), filtered.data());
-            oiio::ImageBufAlgo::median_filter(outBuf, inBuf, medianFilter);
+            ALICEVISION_LOG_ERROR("The output SfMData file '" << sfmOutputDataFilepath << "' cannot be written.");
+            return EXIT_FAILURE;
+        }
+    }
+    else
+    {
+        const fs::path inputPath(inputExpression);
+        std::vector<std::string> filesStrPaths;
 
-            image.swap(filtered);
+        // If sfmInputDataFilename is empty use imageFolders instead
+        if(inputExpression.empty())
+        {
+            for(const std::string& folder : inputFolders)
+            {
+                // If one of the paths isn't a folder path
+                if(!fs::is_directory(folder))
+                {
+                    ALICEVISION_LOG_ERROR("the path '" << folder << "' is not a valid folder path.");
+                    return EXIT_FAILURE;
+                }
+
+                for(fs::directory_entry& entry : fs::directory_iterator(folder))
+                {
+                    const std::string entryPath = entry.path().generic_string();
+                    const std::string ext = entry.path().extension().string();
+                    if(image::isSupported(ext))
+                        filesStrPaths.push_back(entryPath);
+                }
+            }
         }
-        if (sharpenWidth >= 3.f && sharpenContrast > 0.f)
+        else
         {
-            image::Image<image::RGBfColor> filtered(image.Width(), image.Height());
-            const oiio::ImageBuf inBuf(oiio::ImageSpec(image.Width(), image.Height(), nchannels, oiio::TypeDesc::FLOAT), image.data());
-            oiio::ImageBuf outBuf(oiio::ImageSpec(image.Width(), image.Height(), nchannels, oiio::TypeDesc::FLOAT), filtered.data());
-            oiio::ImageBufAlgo::unsharp_mask(outBuf, inBuf, "gaussian", sharpenWidth, sharpenContrast, sharpenThreshold);
+            // If you try to use both a regex-like filter expression and folders as input
+            if(!inputFolders.empty())
```

Comment:
Keep the scope around the log (as it's a define).
```
if(!inputFolders.empty())
{
    ALICEVISION_LOG_WARNING...
}
```

Reply:
Fixed in https://github.com/alicevision/AliceVision/pull/806/commits/f7584167ca69f68bb430cb92284b5733df155fd7 .